airflow/metrics/statsd_logger.py

```diff
@@ -44,27 +46,33 @@
 log = logging.getLogger(__name__)
 
 
-def prepare_stat_with_tags(fn: T) -> T:
-    """Add tags to stat with influxdb standard format if influxdb_tags_enabled is True."""
+def prepare_metric_name_with_tags(fn: T) -> T:
+    """Add tags to metric_name with InfluxDB standard format if influxdb_tags_enabled is True."""
 
     @wraps(fn)
     def wrapper(
-        self, stat: str | None = None, *args, tags: dict[str, str] | None = None, **kwargs
+        self, metric_name: str | None = None, tags: dict[str, str] | None = None
     ) -> Callable[[str], str]:
-        if self.influxdb_tags_enabled:
-            if stat is not None and tags is not None:
-                for k, v in tags.items():
-                    if self.metric_tags_validator.test(k):
-                        if all(c not in [",", "="] for c in f"{v}{k}"):
-                            stat += f",{k}={v}"
-                        else:
-                            log.error("Dropping invalid tag: %s=%s.", k, v)
-        return fn(self, stat, *args, tags=tags, **kwargs)
+        if metric_name is None:
+            metric_name = ""
+
+        if self.influxdb_tags_enabled and tags:
+            valid_tags: dict[str, str] = {}
+
+            for k, v in tags.items():
+                if self.metric_tags_validator.test(k):
+                    if all(c not in [",", "="] for c in v) and all(c not in [",", "="] for c in k):
```

Comment:
```suggestion
                   if all(c not in [",", "="] for c in f"{v}{k}"):
```
Since we are checking if those symbols are in either k or v, then checking if it is in f'{k}{v}' works just fine and looks much neater.